pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/RemoteClient.java

```diff
@@ -264,21 +270,31 @@ public DataInputStream query( String sql, StreamingMode windowMode, long windowS
     return null;
   }
 
+
+  @Override public List<String> getServiceNames() throws SQLException {
+    return getServiceNames( null );
+  }
+
+
   @Override public List<String> getServiceNames( String serviceName ) throws SQLException {
-    return getServiceNames();
+    return getServices( serviceName );
   }
 
-  @Override public List<String> getServiceNames() throws SQLException {
+  private List<String> getServices( String serviceName ) throws SQLException {
     List<String> serviceNames = new ArrayList<String>();
     try {
-      String result = execService( "/listServices" );
+      StringBuilder serviceArguments = new StringBuilder().append( "/listServices" );
+      if ( StringUtils.isNotBlank( serviceName ) ) {
+        serviceArguments.append( "?serviceName=" );
+        serviceArguments.append( serviceName );
```

Comment:
Do a URL encoding of this value. The service name can, for example, have spaces and/or special characters.